moxin-backend/src/backend_impls.rs

```diff
@@ -0,0 +1,716 @@
+            &self,
+            data: ChatRequestData,
+            tx: Sender<anyhow::Result<ChatResponse>>,
+        ) -> bool {
+            self.model_tx.send((data, tx)).is_ok()
+        }
+
+        pub fn stop(self) {
+            let Self {
+                model_tx,
+                model_thread,
+            } = self;
+            drop(model_tx);
+            let _ = model_thread.join();
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+enum FileManagementCommand {
+    GetFeaturedModels(Sender<anyhow::Result<Vec<Model>>>),
+    SearchModels(String, Sender<anyhow::Result<Vec<Model>>>),
+    DownloadFile(FileID, Sender<anyhow::Result<FileDownloadResponse>>),
+    GetDownloadedFiles(Sender<anyhow::Result<Vec<DownloadedFile>>>),
+}
+
+#[derive(Clone, Debug)]
+enum ModelManagementCommand {
+    LoadModel(
+        FileID,
+        LoadModelOptions,
+        Sender<anyhow::Result<LoadModelResponse>>,
+    ),
+    EjectModel(Sender<anyhow::Result<()>>),
+    Chat(ChatRequestData, Sender<anyhow::Result<ChatResponse>>),
+    StopChatCompletion(Sender<anyhow::Result<()>>),
+    // Command to start a local server to interact with chat models
+    StartLocalServer(
+        LocalServerConfig,
+        Sender<anyhow::Result<LocalServerResponse>>,
+    ),
+    // Command to stop the local server
+    StopLocalServer(Sender<anyhow::Result<()>>),
+}
+
+#[derive(Clone, Debug)]
+enum BuiltInCommand {
+    File(FileManagementCommand),
+    Model(ModelManagementCommand),
+}
+
+impl From<Command> for BuiltInCommand {
+    fn from(value: Command) -> Self {
+        match value {
+            Command::GetFeaturedModels(tx) => {
+                Self::File(FileManagementCommand::GetFeaturedModels(tx))
+            }
+            Command::SearchModels(request, tx) => {
+                Self::File(FileManagementCommand::SearchModels(request, tx))
+            }
+            Command::DownloadFile(file_id, tx) => {
+                Self::File(FileManagementCommand::DownloadFile(file_id, tx))
+            }
+            Command::GetDownloadedFiles(tx) => {
+                Self::File(FileManagementCommand::GetDownloadedFiles(tx))
+            }
+            Command::LoadModel(file_id, options, tx) => {
+                Self::Model(ModelManagementCommand::LoadModel(file_id, options, tx))
+            }
+            Command::EjectModel(tx) => Self::Model(ModelManagementCommand::EjectModel(tx)),
+            Command::Chat(request, tx) => Self::Model(ModelManagementCommand::Chat(request, tx)),
+            Command::StopChatCompletion(tx) => {
+                Self::Model(ModelManagementCommand::StopChatCompletion(tx))
+            }
+            Command::StartLocalServer(config, tx) => {
+                Self::Model(ModelManagementCommand::StartLocalServer(config, tx))
+            }
+            Command::StopLocalServer(tx) => {
+                Self::Model(ModelManagementCommand::StopLocalServer(tx))
+            }
+        }
+    }
+}
+
+#[test]
+fn test_chat() {
```

Comment:
This test demonstrates the basic usage, and if I have any misunderstandings regarding the parameters, they can be pointed out here.